### PR TITLE
Gp 23 fe login page

### PR DESCRIPTION
**Pull request by thuanvmse161224**
Adding facebook and google button login

### CODE REVIEW

File: src/pages/Login/assets/FacebookLogo.tsx
Comment (dg-panh) [inline src/pages/Login/assets/FacebookLogo.tsx:1]: Why dont save logo at public/assets folder?

File: src/pages/Login/LoginButton/styles.ts
Comment (dg-panh) [inline src/pages/Login/LoginButton/styles.ts:1]: Already have theme.palette in theme folder. Should declare color here: 
image

File: src/pages/Login/LoginButton/LoginButton.tsx
Comment (dg-panh) [inline src/pages/Login/LoginButton/LoginButton.tsx:1]: Should put simple button in src/component folder and add commonly used variables like button's color, button's title etc.

File: src/pages/Login/FacebookLoginButton/FacebookLoginButton.tsx
Comment (dg-panh) [inline src/pages/Login/FacebookLoginButton/FacebookLoginButton.tsx:1]: Facebook Btn and Google Btn both the same code base, the only difference is the logo and title. Why must create 2 different component. Suggestion: Just create 1 component and pass props(logo, title, ...)

File: src/pages/DevNav/DevNav.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { Box, TextField } from '@mui/material'
+import { Link } from 'react-router-dom'
+
+
+export const DevNav: React.FC = () => {
+	return (
+		<>
+			<Box>
+				<Link to="/login">Login page</Link> <br/>
```
Comment (danielNg91) [inline src/pages/DevNav/DevNav.tsx]: we already have a function to get page link, reuse this function

File: src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -1,10 +1,67 @@
 import React from 'react'
+import { Box, Container, Typography } from '@mui/material'
+import { styled } from '@mui/material/styles'
+import Divider from '@mui/material/Divider'
+import { Link } from 'react-router-dom'
+
+
+import LoginForm from './LoginForm'
+import GoogleLoginButton from './GoogleLoginButton/GoogleLoginButton'
```
Comment (danielNg91) [inline src/pages/Login/Login.tsx]: import from index file

File: src/pages/Login/Login.tsx [hunk 3]
```diff
@@ -1,10 +1,67 @@
...
+import FacebookLoginButton from './FacebookLoginButton/FacebookLoginButton'
```
Comment (danielNg91) [inline src/pages/Login/Login.tsx]: import from index file

File: src/pages/Login/LoginForm.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		type="email"
+		{...register('email', {
+			required: 'Email is Required',
+			pattern: {
+				// eslint-disable-next-line no-useless-escape
+				value: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+				message: 'Not a valid email address'
+			}
+		})}
+		error={!!errors.email}
```
Comment (danielNg91) [inline src/pages/Login/LoginForm.tsx]: use Boolean(errors.email) to get boolean value, remove not not assertion

File: src/pages/Login/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+		margin="normal"
+		type="password"
+		{...register('password', {
+			required: 'Password is Required',
+			pattern: {
+				value: /^(?=.*[A-Za-z])(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{6,20}$/,
+				message: 'Password must start with a letter, contain at least 1 uppercase letter, 1 lowercase letter, and a number. Length should be 6-20 characters.'
+			}
+		})}
+		error={!!errors.password}
```
Comment (danielNg91) [inline src/pages/Login/LoginForm.tsx]: same as above

**Review by danielNg91 — CHANGES_REQUESTED** [review]

File: src/pages/Login/LoginForm.tsx (hunk 5 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm.tsx]: Has fix according to the reply

File: src/pages/Login/Login.tsx (hunk 3 above)
Comment (thuanvmse161224) [reply src/pages/Login/Login.tsx]: import { Button as FacebookLoginBtn, Button as GoogleLoginBtn } from '../../components/LoginWithExternalSiteButton'

Fix

File: src/pages/Login/FacebookLoginButton/FacebookLoginButton.tsx
Comment (thuanvmse161224) [reply src/pages/Login/FacebookLoginButton/FacebookLoginButton.tsx:1]: fixed. these two are now calls the same component with different props

File: src/pages/Login/LoginButton/LoginButton.tsx
Comment (thuanvmse161224) [reply src/pages/Login/LoginButton/LoginButton.tsx:1]: fixed. refer to PublicFormButton for usage with forms

File: src/pages/Login/LoginButton/styles.ts
Comment (thuanvmse161224) [reply src/pages/Login/LoginButton/styles.ts:1]: added orange palette array to palette.ts

File: src/pages/Login/assets/FacebookLogo.tsx
Comment (thuanvmse161224) [reply src/pages/Login/assets/FacebookLogo.tsx:1]: as these two are not pure svg , aka it's actually react component.
needing feedback as to put it in src/assets , public/assets or login/assets

File: src/pages/DevNav/DevNav.tsx (hunk 1 above)
Comment (thuanvmse161224) [reply src/pages/DevNav/DevNav.tsx]: Deleted DevNav . Not needed anymore (chore)

File: src/components/LoginWithExternalSiteButton/styles.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { createTheme } from '@mui/material/styles';
+
+import palette from '../../theme/palette';
+
+
+let themeCreate = createTheme({
```
Comment (danielNg91) [inline src/components/LoginWithExternalSiteButton/styles.ts:6]: define themeCreate as const

File: src/components/PublicFormButton/styles.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { createTheme } from '@mui/material/styles';
+
+import palette from '../../theme/palette';
```
Comment (danielNg91) [inline src/components/PublicFormButton/styles.ts:3]: same as above

File: src/pages/Login/LoginForm/EmailTextField.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+		id="email-input"
+		label="Email Address"
+		variant="outlined"
+		margin="normal"
+		type="email"
+		{...register('email', {
+			required: 'Email is Required',
+			pattern: {
+				// eslint-disable-next-line no-useless-escape
+				value: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/EmailTextField.tsx]: define a const such as: EMAIL_REGEX = /.../, then pass to this value

File: src/pages/Login/LoginForm/LoginForm.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+import {Button as LoginButton} from '../../../components/PublicFormButton'
+import palette from '../../../theme/palette'
+
+import { theme } from './styles'
+import { EmailTextField } from './EmailTextField'
+import { PasswordTextField } from './PasswordTextField'
+import { RememberMeCheckbox } from './RememberMeCheckbox'
+import { LoginInput } from './LoginInput'
+//HACK Where should I put LoginInput Type to ?
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/LoginForm.tsx]: ??

File: src/pages/Login/LoginForm/LoginInput.tsx [hunk 10]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/LoginInput.tsx]: create types.ts in  src/pages/Login/LoginForm, and put types that you need use in elsewhere. Don't remember export it from src/pages/Login/LoginForm/index.ts

File: src/pages/Login/LoginForm/PasswordTextField.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+		label="Password"
+		variant="outlined"
+		margin="normal"
+		type="password"
+		{...register('password', {
+			required: 'Password is Required',
+			pattern: {
+				value: /^(?=.*[A-Za-z])(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{6,20}$/,
+				message: 'Password must start with a letter, contain at least 1 uppercase letter, 1 lowercase letter, and a number. Length should be 6-20 characters.'
+				//FIXME: Fixing this regex bug. It can start with a number, can contain special characters.
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/PasswordTextField.tsx:22]: ??

File: src/pages/Login/LoginForm/PasswordTextField.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+	<TextField
+		id="password-input"
+		label="Password"
+		variant="outlined"
+		margin="normal"
+		type="password"
+		{...register('password', {
+			required: 'Password is Required',
+			pattern: {
+				value: /^(?=.*[A-Za-z])(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{6,20}$/,
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/PasswordTextField.tsx]: define this regex as const same as above

File: src/pages/Login/LoginForm/styles.ts [hunk 13]
```diff
@@ -0,0 +1,19 @@
+import { createTheme } from '@mui/material/styles';
+
+import palette from '../../../theme/palette';
+
+
+let themeCreate = createTheme({
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/styles.ts:6]: define as cont

File: src/theme/palette.ts [hunk 14]
```diff
@@ -15,6 +15,20 @@ const GREY = {
 	900: '#161C24',
 };
 
+// SETUP COLORS
+const ORANGE = {
```
Comment (danielNg91) [inline src/theme/palette.ts]: using Object.freeze make this object immutable in runtime

**Review by danielNg91 — CHANGES_REQUESTED** [review]

File: src/components/LoginWithExternalSiteButton/styles.ts (hunk 6 above)
Comment (thuanvmse161224) [reply src/components/LoginWithExternalSiteButton/styles.ts:6]: export const theme = themeCreate;
it needs to be declared before using augmentColor function, so it's necessary to using 'let' keyword

File: src/components/PublicFormButton/styles.ts (hunk 7 above)
Comment (thuanvmse161224) [reply src/components/PublicFormButton/styles.ts:3]: export const theme = themeCreate;

File: src/pages/Login/LoginForm/EmailTextField.tsx (hunk 8 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/EmailTextField.tsx]: will work on it

File: src/pages/Login/LoginForm/LoginForm.tsx (hunk 9 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/LoginForm.tsx]: it's a call for attention, so it served its purpose

File: src/pages/Login/LoginForm/LoginInput.tsx (hunk 10 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/LoginInput.tsx]: thanks

File: src/pages/Login/LoginForm/PasswordTextField.tsx (hunk 11 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/PasswordTextField.tsx:22]: I'm not sure about how the password would be processing in to back-end. 
'Password must start with a letter, contain at least 1 uppercase letter, 1 lowercase letter, and a number. Length should be 6-20 characters.'
But, testing with real string:
-  57Agms is allowed
- 57Agms%$ is allowed
Two of the above shouldn't happened if password must abide to the law specified above.
Feedback on how this regex should be thanks

File: src/pages/Login/LoginForm/PasswordTextField.tsx (hunk 12 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/PasswordTextField.tsx]: it's easy enough to do. work on it soon

File: src/pages/Login/LoginForm/styles.ts (hunk 13 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/styles.ts:6]: export const theme = themeCreate;

File: src/theme/palette.ts (hunk 14 above)
Comment (thuanvmse161224) [reply src/theme/palette.ts]: If so, should other constant in palette.ts should be freeze as well ?

**Review by dg-panh — COMMENTED** [review]
LGTM

File: src/theme/palette.ts (hunk 14 above)
Comment (danielNg91) [reply src/theme/palette.ts]: just use when you need

File: src/pages/Login/LoginForm/styles.ts (hunk 13 above)
Comment (danielNg91) [reply src/pages/Login/LoginForm/styles.ts:6]: ok fine

File: src/pages/Login/LoginForm/PasswordTextField.tsx (hunk 11 above)
Comment (danielNg91) [reply src/pages/Login/LoginForm/PasswordTextField.tsx:22]: should have simple verification in fe, so could you change the regex to satisfy ''Password must start with a letter, contain at least 1 uppercase letter, 1 lowercase letter, and a number. Length should be 6-20 characters.''

File: src/pages/Login/assets/FacebookLogo.tsx
Comment (dg-panh) [reply src/pages/Login/assets/FacebookLogo.tsx:1]: I'm also wondering whether to put the assets in public or in the src @@ but I already use public/assets

File: src/pages/Login/LoginForm/PasswordTextField.tsx (hunk 12 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/PasswordTextField.tsx]: done

File: src/pages/Login/LoginForm/PasswordTextField.tsx (hunk 11 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/PasswordTextField.tsx:22]: fix bug successfully

File: src/pages/Login/LoginForm/EmailTextField.tsx [hunk 15]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { TextField } from '@mui/material'
+import { UseFormRegister, FieldErrors } from 'react-hook-form'
+
+import { LoginInputType } from './types'
+
+
+// eslint-disable-next-line no-useless-escape
+const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/EmailTextField.tsx]: should be UPPER_CASE_SNAKE_CASE for constant

File: src/pages/Login/LoginForm/index.ts [hunk 16]
```diff
@@ -0,0 +1,2 @@
+export {default as LoginForm} from './LoginForm';
+export type {LoginInputType} from './types';
```
Comment (danielNg91) [inline src/pages/Login/LoginForm/index.ts]: just recommend, export *  from './types'

File: src/theme/palette.ts [hunk 17]
```diff
@@ -15,6 +15,20 @@ const GREY = {
...
+	300: '#f0b850',
+	400: '#eda92c',
+	500: '#ec9b0e',
+	600: '#e88f09',
+	700: '#e28004',
+	800: '#dc7200',
+	900: '#d25a00',
+});
+
 const PRIMARY = {
```
Comment (danielNg91) [inline src/theme/palette.ts]: Object.freeze

File: src/theme/palette.ts [hunk 18]
```diff
@@ -25,12 +39,12 @@ const PRIMARY = {
 };
 
 const SECONDARY = {
```
Comment (danielNg91) [inline src/theme/palette.ts]: Object.freeze

**Review by danielNg91 — CHANGES_REQUESTED** [review]

File: src/pages/Login/LoginForm/index.ts (hunk 16 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/index.ts]: thx. done

File: src/pages/Login/LoginForm/EmailTextField.tsx (hunk 15 above)
Comment (thuanvmse161224) [reply src/pages/Login/LoginForm/EmailTextField.tsx]: renamed const to correct convention

File: src/pages/Login/assets/FacebookLogo.tsx
Comment (thuanvmse161224) [reply src/pages/Login/assets/FacebookLogo.tsx:1]: Done. Go to public / icons

**Review by danielNg91 — APPROVED** [review]

File: src/pages/Login/assets/FacebookLogo.tsx
Comment (thuanvmse161224) [reply src/pages/Login/assets/FacebookLogo.tsx:1]: has put it to public/assets/icons

**Pull request merged**